# HG changeset patch
# Parent c640573a321fd425a7fd4533264001655e498330
# User sbr
# Date 1347351856 -7200

-Fix [FS#5203] (r24521): Tests first next character before removing it.

diff -r c640573a321f src/textbuf.cpp
--- a/src/textbuf.cpp	Mon Sep 10 18:47:13 2012 +0000
+++ b/src/textbuf.cpp	Tue Sep 11 10:26:08 2012 +0200
@@ -113,9 +113,6 @@
 
 		if (!CanDelChar(backspace)) return false;
 
-		/* Unconditionally delete one char to the left. */
-		this->DelChar(backspace);
-		if (!CanDelChar(backspace)) return false;
 		WChar c = this->GetNextDelChar(backspace);
 
 		/* Backspace: Delete left whitespaces.
